pybtp/btp.py

```diff
@@ -604,24 +615,34 @@ def gap_adv_ind_on(ad={}, sd={}, duration=AdDuration.forever, own_addr_type=OwnA
     ad_ba = bytearray()
     sd_ba = bytearray()
 
-    for ad_type, ad_data in ad.iteritems():
-        data = binascii.unhexlify(bytearray(ad_data))
-        ad_ba.extend(chr(ad_type))
-        ad_ba.extend(chr(len(data)))
+    for ad_type, ad_data in list(ad.items()):
+        if isinstance(ad_data, str):
+            data = bytes.fromhex(ad_data.decode('utf-8'))
```

Comment:
if `isinstance(ad_data, str)` then, to the best of my understanding, Python3 does not provide `str.decode()`

Reply:
Thank you for catching this! Yes, this should be correctly formatted string, and indeed these neither have `decode()` method, nor they need it.